### PR TITLE
Support for paid course assessments

### LINKED ISSUE #6473: Ceepos-verkkomaksu: Aineopiskelu/lukio
Ceepos-verkkomaksun laajentaminen kurssiperusteiseen laskutukseen (Aineopiskelu/lukio).
Tämä vaatii oletettavasti enemmän suunnittelua kuin yksityisopiskelija-linjan maksu.

### PR DESCRIPTION
**Pull request by pkukkon**
- resolves #6473 
- requires Pyramus issue 1501
- correct database version still needs to be determined

### CODE REVIEW

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/ceepos/rest/CeeposRESTService.java [hunk 1]
```diff
@@ -752,6 +781,59 @@ public Response removeeOrder(@PathParam("ORDERID") Long orderId) {
     return Response.noContent().build();
   }
   
+  @Path("/order/{ORDERID}/returnLink")
+  @GET
+  @RESTPermit(handling = Handling.INLINE)
+  public Response getReturnLink(@PathParam("ORDERID") Long orderId) {
```
Comment (aviljakaine) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/ceepos/rest/CeeposRESTService.java:787]: Should this have a check to make sure the logged user is the owner of the order or something?

**Review by aviljakaine — COMMENTED** [review]
- Figure out the correct db updater script version.
- Consider additional check
- Otherwise backend ok

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/ceepos/rest/CeeposRESTService.java (hunk 1 above)
Comment (pkukkon) [reply muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/ceepos/rest/CeeposRESTService.java:787]: Added a safety check for order ownership.

**Review by aviljakaine — COMMENTED** [review]
Backend ok, needs database script version though

**Review by saaralepisto — APPROVED** [review]

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/base/workspace/evaluation-request-dialog.tsx [hunk 2]
```diff
@@ -94,33 +104,133 @@ class EvaluationRequestDialog extends React.Component<
...
+          }
+        ),
+        "callback"
+      )().then((link: { url: string }) => link.url)) as string;
+      window.location.href = link;
+    } catch (e) {
+      this.props.t("notifications.createError", {
+        ns: "orders",
+        context: "paymentLocation",
+      });
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/base/workspace/evaluation-request-dialog.tsx:128]: There probably should displayNotification?

**Review by TeroOtavia — CHANGES_REQUESTED** [review]

**Review by aviljakaine — DISMISSED** [review]
Backend ok

**Review by saaralepisto — DISMISSED** [review]

**Review by TeroOtavia — APPROVED** [review]

**Pull request merged**